desc/objectives/_neoclassical.py

```diff
@@ -0,0 +1,253 @@
+        desired bound (lower or upper) of the profile at those locations.
+        Defaults to ``bounds=(0, np.inf)``
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute Jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    grid : Grid, optional
+        Collocation grid to evaluate flux surface averages.
+        Should have poloidal and toroidal resolution.
+    alpha : ndarray
+        Unique coordinate values for field line poloidal angle label alpha.
+    zeta : ndarray
+        Unique coordinate values for field line following coordinate zeta. Must be
+        strictly increasing. A good reference density is 100 knots per toroidal transit.
+        For axisymmetric devices only one toroidal transit is necessary. Otherwise,
+        more toroidal transits will give more accurate result, with diminishing returns.
+    num_quad : int
+        Resolution for quadrature of bounce integrals. Default is 31,
+        which gets sufficient convergence, so higher values are likely unnecessary.
+    num_pitch : int
+        Resolution for quadrature over velocity coordinate, preferably odd.
+        Default is 99. Effective ripple will look smoother at high values.
+        (If computed on many flux surfaces and micro oscillation is seen
+        between neighboring surfaces, increasing num_pitch will smooth the profile).
+    batch : bool
+        Whether to vectorize part of the computation. Default is true.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    _coordinates = "r"
+    _units = "~"
+    _print_value_fmt = "Effective ripple ε¹ᐧ⁵: {:10.3e} "
+
+    def __init__(
+        self,
+        eq,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        grid=None,
+        alpha=np.array([0]),
```

Comment:
or use more alpha values?

Reply:
So in general, using more alpha values is not a replacement for a longer field line. Some bounce integrals are multiple toroidal transits wide which will not be captured if the field line is truncated. (We can make it equivalent, if we stitch together the field lines to whatever their new alpha value is at the zeta which the field line is truncated. Then we add up the integrals that got split up).

So as long as the rotational transform is such that a single field line can cover the flux surface it's better to just have one alpha and a long field line. if the field line closes on itself quickly, or stays in a small strip (think these are called near rational surfaces) then multiple alphas are better.